GameDesign/Assets/Scripts/MonoBehaviours/CustomerMonoBehavior.cs

```diff
@@ -0,0 +1,75 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CustomerMonoBehavior : MonoBehaviour, Clickable
+{
+    [SerializeField] public int id;
+    [SerializeField] public float baseSpeed;
+    [SerializeField] public float currentSpeed;
+    [SerializeField] public int clickAmount = 0;
+    [SerializeField] public int clickTime;
+    [SerializeField] public bool wearsMask;
+    [SerializeField] public NPCMovementManager moveManager;
+    [SerializeField] public int pointValue;
+    [SerializeField] public bool isFrozen = false;
+    [SerializeField] public int frozenTimeCount = 0;
+
+    [SerializeField] public TaserManager taserManager = TaserManager.Instance;
+
+    void Start()
+    {
+        InvokeRepeating("UnFreeze", 1.0f, 1.0f);
```

Comment:
Shouldn't we activate this invoke only when we freeze it instead of everytime? (And deactivate it when it gets unfrozen)

Reply:
i think it could be implemented as a state or a coroutine as a better practice, personally i think the state may be better. and we just keep a variable maskable:bool  to say if shooting it with a mask activates one behavior or another.

Reply:
It didn't work, because the InvokeRepeating was finishing when the where it was called finished. Couldn't fine alternatives for timed counters.